lib/Doctrine/Common/Annotations/FileCacheReader.php

```diff
@@ -214,14 +214,14 @@ private function saveCacheFile($path, $data)
             throw new \RuntimeException(sprintf('Unable to create tempfile in directory: %s', $this->dir));
         }
 
+        @chmod($tempfile, 0666 & (~$this->umask));
+
         $written = file_put_contents($tempfile, '<?php return unserialize('.var_export(serialize($data), true).');');
 
         if (false === $written) {
             throw new \RuntimeException(sprintf('Unable to write cached file to: %s', $tempfile));
         }
 
-        @chmod($tempfile, 0666 & (~$this->umask));
```

Comment:
Either that, or do chmod twice. At the end of the process, the flags on the file should be `0666 & (~$this->umask)`.